destination/franz.go

```diff
@@ -36,13 +45,32 @@ func NewFranzProducer(ctx context.Context, cfg Config) (*FranzProducer, error) {
 	opts := cfg.FranzClientOpts(sdk.Logger(ctx))
 	opts = append(opts, []kgo.Opt{
 		kgo.AllowAutoTopicCreation(),
-		kgo.DefaultProduceTopic(cfg.Topic),
 		kgo.RecordDeliveryTimeout(cfg.DeliveryTimeout),
 		kgo.RequiredAcks(cfg.RequiredAcks()),
 		kgo.ProducerBatchCompression(cfg.CompressionCodecs()...),
 		kgo.ProducerBatchMaxBytes(cfg.BatchBytes),
 	}...)
 
+	var topicFn func(sdk.Record) (string, error)
+	if strings.Contains(cfg.Topic, "{{") && strings.Contains(cfg.Topic, "}}") {
```

Comment:
I wonder if we could instead try to parse it and if we can't, then directly do `opts = append(opts, kgo.DefaultProduceTopic(cfg.Topic))`. My point is that, even though the topic could _look like_ a template, we still need to parse it like we do below.

Reply:
Good point 👍

Reply:
Actually, a string that doesn't contain a go template like `foo` is a valid go template. But I'd like to avoid creating and executing a Go template if it's only a static string.

Regardless, I'll move some things around, because we should validate that the topic is valid in `Configure` and not in `Open`.